src/libs/antares/study/parts/thermal/cluster.cpp

```diff
@@ -434,6 +474,72 @@ void Data::ThermalCluster::calculationOfSpinning()
     }
 }
 
+void Data::ThermalCluster::calculationOfMarketBidPerHourAndMarginalCostPerHour()
+{
+    if (costgeneration == Data::setManually || !prepro || efficiency <= 0.0)
+    {
+        std::fill(
+          marketBidCostPerHourTs[0].begin(), marketBidCostPerHourTs[0].end(), marketBidCost);
+        std::fill(marginalCostPerHourTs[0].begin(), marginalCostPerHourTs[0].end(), marginalCost);
+        return;
+    }
+    else // costgeneration == Data::useCostTimeseries
+    {
+        for (uint hour = 0; hour < HOURS_PER_YEAR; ++hour)
+        {
+            marketBidCostPerHourTs[0][hour] = prepro->fuelcost[0][hour] * 360.0 / efficiency
+                                              + co2 * prepro->co2cost[0][hour] + variableomcost;
+            marginalCostPerHourTs[0][hour] = marketBidCostPerHourTs[0][hour];
+            if (modulation.width > 0) // we should update production cost when modulation is ready
+            {
+                productionCostTs[0][hour]
+                  = marginalCostPerHourTs[0][hour] * modulation[Data::thermalModulationCost][hour];
+            }
+        }
+
+        // calculate marketBidCost and marginalBidCost as average of the first column
+        marketBidCost
+          = std::accumulate(marketBidCostPerHourTs[0].begin(), marketBidCostPerHourTs[0].end(), 0)
+            / HOURS_PER_YEAR;
+        marginalCost
+          = std::accumulate(marginalCostPerHourTs[0].begin(), marginalCostPerHourTs[0].end(), 0)
+            / HOURS_PER_YEAR;
+
+        // multiple TS
+        uint fuelCostWidth = prepro->fuelcost.width;
+        uint co2CostWidth = prepro->co2cost.width;
+        uint tsCount = Math::Max(fuelCostWidth, co2CostWidth);
+
+        // if fuel and co2 cost have only one TS -> exit here
+        if (tsCount == 1)
+            return;
+        
+        // if fuel or co2 cost have more than one TS -> continue    
+        for (uint tsIndex = 2; tsIndex <= tsCount; ++tsIndex)
+        {
+            uint tsIndexFuel = Math::Min(fuelCostWidth, tsIndex);
+            uint tsIndexCo2 = Math::Min(co2CostWidth, tsIndex);
+            marketBidCostPerHourTs.push_back(tmp); // add blank array with 8760-zeros
+            marginalCostPerHourTs.push_back(tmp);  // add blank array with 8760-zeros
+            productionCostTs.push_back(tmp);  // add blank array with 8760-zeros
```

Comment:
You can call [`std::vector<>::resize`](https://cplusplus.com/reference/vector/vector/resize/). For example

```cpp
marketBidCostPerHourTs.resize(tsCount, tmp);
```

In fact, `tmp` may not even be required since `resize` calls the default constructor for `std::array`, which gives `{0, ...,0}`.